easyrest/models/user.py

```diff
@@ -64,9 +74,12 @@ def add(database, form_data):
             "required": ["name", "email", "password"]
         }
         validation(schema, form_data)
-        name = form_data['name']
         email = form_data['email']
+        email_in_database = database.query(User.email).filter(User.email == email).scalar()
+        if email_in_database is not None:
+            raise ValidationError('already exist', email)
+        name = form_data['name']
         password = form_data['password']
         password_hash = pbkdf2_sha256.hash(password)
 
-        database.add(User(name=name, email=email, password=password_hash))
+        database.add(User(name=name, email=email, password=password_hash, role_id=role))
```

Comment:
Here you assign role by hard coding id of role_name from another table. Problem is that if we change initialization script and roles changed their id you will need to change constants accordingly to new ids. Maybe better do another query to get id of role. Like: 
role_model = database.query(UserRole).filter_by(name == role_name).first()
user = User(name=name, email=email, password=password_hash)
user.role = role_model
Or you can add 1 more field to existing query:
email, role = database.query(User.email, UserRole).filter(User.email == email, UserRole.name == role_name).first()

Reply:
You say that id will change, but for some reason do not say that the name of the role may change.
1. Yes, I use constants. And I use the same constants in `user_controller.py` and it increases the readability of the code:
_user_controller.py line 164_
   `if new_user_role == User.CLIENT`
Not bad, right?
2. Making a request to the database takes much more time than getting a constant value.
3. At the stage of database design, Sergey said approximately the following:
No need to create extra tables and relations to them if they very rarely change.
It was when there was a discussion of the status of orders, restaurants, etc. Remember these separate tables with three or two records?

I still don’t understand why it was necessary to create a table `user_roles` + relation, if relatively static data is stored in it.